frontend/src/lib/components/PropertyFilters/components/TaxonomicPropertyFilter/infiniteListLogic.ts

```diff
@@ -0,0 +1,204 @@
+    props: {} as TaxonomicPropertyFilterListLogicProps,
+
+    key: (props) => `${props.pageKey}-${props.filterIndex}-${props.type}`,
+
+    connect: (props: TaxonomicPropertyFilterListLogicProps) => ({
+        values: [taxonomicPropertyFilterLogic(props), ['searchQuery']],
+        actions: [taxonomicPropertyFilterLogic(props), ['setSearchQuery']],
+    }),
+
+    actions: {
+        setLimit: (limit: number) => ({ limit }),
+        onRowsRendered: (rowInfo: RenderedRows) => ({ rowInfo }),
+        loadRemoteItems: (options: LoaderOptions) => options,
+    },
+
+    reducers: () => ({
+        limit: [
+            100,
+            {
+                setLimit: (_, { limit }) => limit,
+            },
+        ],
+    }),
+
+    loaders: ({ values }) => ({
+        remoteItems: [
+            createEmptyListStorage('', true),
+            {
+                loadRemoteItems: async ({ offset, limit }, breakpoint) => {
+                    // avoid the 150ms delay on first load
+                    if (!values.remoteItems.first) {
+                        await breakpoint(150)
+                    }
+
+                    const { remoteEndpoint, searchQuery } = values
+
+                    if (!remoteEndpoint) {
+                        // should not have been here in the first place!
+                        return createEmptyListStorage(searchQuery)
+                    }
+
+                    const url = combineUrl(remoteEndpoint, {
+                        search: searchQuery,
+                        limit,
+                        offset,
+                    }).url
+
+                    let response: EventDefinitionStorage
+
+                    if (apiCache[url]) {
+                        response = apiCache[url]
+                    } else {
+                        response = await api.get(url)
+                        apiCache[url] = response
+                        apiCacheTimers[url] = window.setTimeout(() => {
+                            delete apiCache[url]
+                            delete apiCacheTimers[url]
+                        }, API_CACHE_TIMEOUT)
+                        breakpoint()
+                    }
+
+                    return {
+                        results: appendAtIndex(
+                            values.items.searchQuery === values.searchQuery ? values.items.results : [],
+                            response.results,
+                            offset
+                        ),
+                        searchQuery: values.searchQuery,
+                        count: response.count,
+                    }
+                },
+            },
+        ],
+    }),
+
+    listeners: ({ values, actions }) => ({
+        onRowsRendered: ({ rowInfo: { startIndex, stopIndex, overscanStopIndex } }) => {
+            if (values.isRemoteDataSource) {
+                let loadFrom: number | null = null
+                for (let i = startIndex; i < (stopIndex + overscanStopIndex) / 2; i++) {
```

Comment:
Won't this load every single time another row is rendered? Fairly certain that `onRowsRendered` is potentially fired many times per second; this will result in a lot of network requests for overlapping data.